arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowTable.java

```diff
@@ -185,18 +188,24 @@ private static RelDataType deduceRowType(Schema schema,
   }
 
   private static TreeNode makeLiteralNode(String literal, String type) {
-    switch (type) {
-    case "integer":
+    if (type.startsWith("decimal")) {
```

Comment:
If we have RelDataType, then we can switch by SqlTypeName and get the precision and scale by RelDataType.

Reply:
It looks a bit cumbersome at the moment, and it's almost like refactoring the filter structure. Another reason is that the parameter of arrow's makeLiteralNode api must be a string, which means I need another layer of conversion, but you can try

Reply:
@NobiGo I have created a Jira case, but I am very busy with work recently and I think I will have to wait until later to start working on this.
https://issues.apache.org/jira/browse/CALCITE-6701

Reply:
this is a private function, why does the parameter have to be a string?

Reply:
@mihaibudiu @NobiGo Hi !
In the `ArrowTranslator `class, we will translate the corresponding filter conditions according to the logical plan and return a string list, such as `decimalField equal 1 decimal(19,0)`. Then I need to parse this string array and execute the corresponding arrow API in arrowtable.
Since the parameters I received in ArrowTable are already string arrays, I can only process strings, which is why makeLiteralNode is all strings.

Reply:
This is a strange design. Turns out that the table holds a program represented as a text that contains the pushdown filters and projections.

Reply:
Yes, I don’t know why I wrote it like this in the beginning. I am thinking about how to optimize this part. It seems to be a very big change.